src/search-injection/content_script.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+
+import Results from './components/Results'
+import * as constants from './constants'
+import { appendCss, getLocalStorage } from './utils'
+import { SEARCH_CONN_NAME, CMDS } from '../overview/constants'
+
+const handleRender = results => {
+    // The actual function to render the results on screen.
+
+    const renderComponent = async () => {
+        const position = await getLocalStorage(constants.POSITION_KEY, 'above')
+        const containerID = constants.SEARCH_ENGINES.google.container[position]
+        // Gets the container using the passed id
+        const container = document.getElementById(containerID)
+
+        // If re-rendering remove the already present component
+        const component = document.getElementById('memexResults')
+        if (component) component.parentNode.removeChild(component)
+
+        const target = document.createElement('div')
+        target.setAttribute('id', 'memexResults')
+        container.insertBefore(target, container.firstChild)
+
+        // Number of results to limit
+        const limit = constants.LIMIT[position]
+
+        // Render our React component on the target element
+        // Passing this same function so that it can change position
+        ReactDOM.render(
+            <Results
+                results={results.slice(0, limit)}
+                len={results.length}
+                rerender={renderComponent}
+            />,
+            target,
+        )
+    }
+
+    // Append content_script.css to the document
+    const cssFile = browser.extension.getURL('/content_script.css')
+    appendCss(cssFile)
```

Comment:
Seems a bit odd to have the `appendCss` logic - which **prepends** (possible naming issue) a `<script>` tag to the DOM - in the `search-injection/utils` module while the other DOM stuff is in this module. Maybe you could move `appendCss` and maybe also have a `appendReactRoot` function to their own `search-injection/dom` module or something. Just a code-layout suggestion; the actual code seems very well done.

Reply:
*prepend* was a very recent change, so forgive me on that. It injects CSS into the page, so I guess `injectCSS` would fit it. And that's a nice idea, I will do it.